.github/workflows/end_to_end_test.yaml

```diff
@@ -0,0 +1,61 @@
+name: End to end tests
+
+on:
+  pull_request:
+  workflow_call:
+  workflow_dispatch:
+
+jobs:
+  e2e-test:
+    name: end to end test
+    runs-on: [self-hosted, linux, x64, e2e-runner]
+    steps:
+      - name: Echo hello world
+        run: echo "hello world"
+      - name: File permission for /usr/local/bin
+        run: ls -ld /usr/local/bin | grep drwxrwxrwx
+      - name: Test file permission for /usr/local/bin
+        run: touch /usr/local/bin/test_file
+      # "Install microk8s" step will test if the proxies settings are correct.
+      - name: Proxy set in /etc/environment
+        run: cat /etc/environment | grep HTTP_PROXY
+      # "Update apt in python docker container" step will test docker default proxy settings due to
+      # pulling the python image.
+      - name: Proxy set in docker daemon
+        run: sudo cat /etc/systemd/system/docker.service.d/http-proxy.conf | grep HTTP_PROXY
+      # "Update apt in python docker container" step will test docker client default proxy settings.
+      - name: Proxy set in docker client
+        run: cat /home/ubuntu/.docker/config.json | grep httpProxy
+      - name: Install microk8s
+        run: sudo snap install microk8s --classic
```

Comment:
Can we install **and add** the current user to the microk8s group, so we don't resort to sudo below?

Reply:
Minor comments

Reply:
I'm pretty sure this is done now